nibabel/streamlines/tck.py

```diff
@@ -0,0 +1,440 @@
+
+        # Build header dictionary from the buffer.
+        hdr = dict(item.split(': ') for item in buf.rstrip().split('\n')[:-1])
+        hdr[Field.MAGIC_NUMBER] = magic_number
+
+        # Check integrity of TCK header.
+        if 'datatype' not in hdr:
+            raise HeaderError("Missing 'datatype' attribute in TCK header.")
+
+        if not hdr['datatype'].startswith('Float32'):
+            msg = ("TCK only supports float32 dtype but 'dataype: {}' was"
+                   " specified in the header.").format(hdr['datatype'])
+            raise HeaderError(msg)
+
+        if 'file' not in hdr:
+            raise HeaderError("Missing 'file' attribute in TCK header.")
+
+        if hdr['file'].split()[0] != '.':
+            msg = ("TCK only supports single-file - in other words the"
+                   " filename part must be specified as '.' but '{}' was"
+                   " specified.").format(hdr['file'].split()[0])
+            raise HeaderError("Missing 'file' attribute in TCK header.")
+
+        # Set endianness and _dtype attributes in the header.
+        hdr[Field.ENDIANNESS] = '<'
+        if hdr['datatype'].endswith('BE'):
+            hdr[Field.ENDIANNESS] = '>'
+
+        hdr['_dtype'] = np.dtype(hdr[Field.ENDIANNESS] + 'f4')
+
+        # Keep the file position where the data begin.
+        hdr['_offset_data'] = int(hdr['file'].split()[1])
+
+        # Set the file position where it was, if it was previously open.
+        if start_position is not None:
+            fileobj.seek(start_position, os.SEEK_SET)
+
+        return hdr
+
+    @staticmethod
+    def _read(fileobj, header, buffer_size=4):
+        """ Return generator that reads TCK data from `fileobj` given `header`
+
+        Parameters
+        ----------
+        fileobj : string or file-like object
+            If string, a filename; otherwise an open file-like object
+            pointing to TCK file (and ready to read from the beginning
+            of the TCK header). Note that calling this function
+            does not change the file position.
+        header : dict
+            Metadata associated with this tractogram file.
+        buffer_size : float, optional
+            Size (in Mb) for buffering.
+
+        Yields
+        ------
+        points : ndarray of shape (n_pts, 3)
+            Streamline points
+        """
+        dtype = header["_dtype"]
+        coordinate_size = 3 * dtype.itemsize
+        # Make buffer_size an integer and a multiple of coordinate_size.
+        buffer_size = int(buffer_size * MEGABYTE)
+        buffer_size += coordinate_size - (buffer_size % coordinate_size)
+
+        with Opener(fileobj) as f:
+            start_position = f.tell()
+
+            # Set the file position at the beginning of the data.
+            f.seek(header["_offset_data"], os.SEEK_SET)
+
+            eof = False
+            buff = b""
+            pts = []
+
+            i = 0
+
+            while not eof or not np.all(np.isinf(pts)):
```

Comment:
I don't understand the `not np.all(np.isinf(pts))` part of this - can you explain?